### PR TITLE
Remove deprecated members from driver-utils

### PR DESCRIPTION
**Pull request by ChumpChief**
This change follows up on the deprecations from #14265, #14276, #14168, #12233, and #11811 by removing or hiding those deprecated items.

~I did not remove the deprecated `IAnyDriverError` in this change as it still has usage in the repo that needs to be sorted out.~

EDIT: I removed the `IAnyDriverError` usage in #14493, so pending that getting ported to next I'll probably try to include it too.

EDIT2: `IAnyDriverError` is now also removed.

### CODE REVIEW

File: packages/tools/fetch-tool/src/fluidFetchSnapshot.ts [hunk 1]
```diff
@@ -290,14 +287,7 @@ export async function fluidFetchSnapshot(documentService?: IDocumentService, sav
 
 	console.log("\n");
 
-	let storage = await documentService.connectToStorage();
-	if (paramUnpackAggregatedBlobs) {
-		storage = BlobAggregationStorage.wrap(
```
Comment (ssimic2) [inline packages/tools/fetch-tool/src/fluidFetchSnapshot.ts:295]: Curious on the reason of this change?

File: packages/drivers/routerlicious-driver/src/cache.ts [hunk 2]
```diff
@@ -3,7 +3,7 @@
  * Licensed under the MIT License.
  */
 
-import { MapWithExpiration } from "@fluidframework/driver-utils";
+import { MapWithExpiration } from "./mapWithExpiration";
```
Comment (ssimic2) [inline packages/drivers/routerlicious-driver/src/cache.ts:6]: Is the intention to fully remove MapWithExpiration or just not export it at this point?

Comment (ChumpChief) [reply packages/drivers/routerlicious-driver/src/cache.ts:6]: Just to not export it.  I believe we have many cache-like behaviors in the repo that we could consider consolidating at some point, but probably not a priority.

**Review by alexvy86 — APPROVED** [review]
image
❤️

**Review by jatgarg — APPROVED** [review]

**Pull request merged**